File: reverseWords.py
class Solution:
    def reverseWords(self, s: str) -> str:
        if len(s) <= 1:
            return s
        
        s = s.split() 
        j = len(s) - 1

        for i in range(j):
            if i >= j:
                break
            s[i], s[j] = s[j], s[i]
            j -= 1
        return ' '.join(s)



obj = Solution()
print(obj.reverseWords("blue is sky the"))
print(obj.reverseWords("abc"))
print(obj.reverseWords(""))
print(obj.reverseWords("Word word"))
print(obj.reverseWords("many       spaces"))

File: Javascript/reverseWords.js
var reverseWords = function(s) {
    let words = s.split(' ');
    let rev_words = [];

    for (let i = words.length - 1; i >= 0; i--) {
        if (words[i] !== '') {
            rev_words.push(words[i])
        }
    };

    return rev_words.join(' ');
};


console.log(reverseWords('the sky is blue'))
console.log(reverseWords('      the sky is blue      '))